### ISSUE TITLE
[Dependency Injection] You have requested a non-existent service "fos_elastica.index_manager".

### ISSUE THREAD

**Issue by bpolaszek** [FriendsOfSymfony/FOSElasticaBundle#1271]
Hello there,

I'm trying to install this bundle over a fresh Symfony installation after not using it for a while.
I followed [the instructions](https://github.com/FriendsOfSymfony/FOSElasticaBundle/blob/master/Resources/doc/setup.md)  for setting up the bundle, but didn't succeed to get over that exception:

```
[Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException]  
  You have requested a non-existent service "fos_elastica.index_manager".
```

Here's how to reproduce:
> composer create-project symfony/framework-standard-edition && cd framework-standard-edition
> _Downloads a fresh copy of Symfony stack 3.3.6_

> composer require friendsofsymfony/elastica-bundle
> _Using version ^4.0 for friendsofsymfony/elastica-bundle_

Then, add `new \FOS\ElasticaBundle\FOSElasticaBundle(),` into the `AppKernel`

Then call `fos:elastica:populate`:

> php bin/console fos:elastica:populate

```
[Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException]  
  You have requested a non-existent service "fos_elastica.index_manager".
```

In Symfony 3.3, services are private by default. I disabled this, but nothing changed. I ever tried with a fresh SF 3.2.13 install, same here.

Is there's something I'm doing wrong? Anyway, I shouldn't face a DependencyInjection issue as soon as the bundle is enabled into the AppKernel, should I?

Thank you,
Ben

**Comment by Filoz** [comment]
I have the same problem... in Symfony 3.2

**Comment by Filoz** [comment]
if you try
`php bin/console debug:container`

can you see any fos:elastica services?
I can't

**Comment by nickel715** [comment]
Had the same issue, adding the client config like this solved the problem for me!

```
fos_elastica:
    clients:
        default: { host: localhost, port: 9200 }
```

**Comment by ajielee** [comment]
Same here, in Symfony 3.1

@nickel715 even I add `clients` setting, it doesn't work

**Comment by ajielee** [comment]
OK, I found the solution, we just need to clear the cache before calling `fos:elastica:populate`:

```
php bin/console cache:clear -e=dev && php bin/console cache:clear -e=prod --no-debug
```

**Comment by Tobion** [comment]
The problem here is that the commands are auto-registered by the Symfony. But they are useless until you define a fos elastica client and index. Only then the services are registered that are needed by the commands.

This problem is fixed by having commands as services since #1331 which are only registered when the fos_elastica configuration is done as well. Also there is a recipe for symfony flex https://github.com/symfony/recipes-contrib/tree/master/friendsofsymfony/elastica-bundle/5.0 which makes this easier to set up.
